lib/Service/SlackAPIService.php

```diff
@@ -171,31 +171,39 @@ public function getMyChannels(string $userId, bool $useCache): array {
 
 		foreach ($rawChannels as $channel) {
 			if (
-				isset(
-					$channel['is_group'],
-					$channel['is_mpim'],
-					$channel['name'],
-					$channel['purpose'],
-					$channel['purpose']['value'],
-					$channel['topic'],
-					$channel['topic']['value']
-				) && ($channel['is_group'] || $channel['is_mpim'])
+				(isset($channel['is_group']) && $channel['is_group'])
+				|| (isset($channel['is_mpim']) && $channel['is_mpim'])
 			) {
-				$groupName = array_filter(
-					[$channel['topic']['value'], $channel['purpose']['value'], $channel['name'], 'Group ' . $channel['id']],
-					fn ($val) => $val !== '' && $val !== null
-				)[0];
+				$groupName = array_values(array_filter(
+					[
+						$channel['topic']['value'] ?? null,
+						$channel['purpose']['value'] ?? null,
+						$channel['name'],
+						'Group ' . $channel['id'],
+					],
+					fn ($val) => $val !== '' && $val !== null,
+				))[0];
 
 				$channels[] = [
 					'id' => $channel['id'],
 					'name' => $groupName,
 					'type' => 'group',
 					'updated' => $channel['updated'] ?? 0,
 				];
-			} elseif (isset($channel['is_channel'], $channel['name']) && $channel['is_channel']) {
+			} elseif (isset($channel['is_channel']) && $channel['is_channel']) {
+				$channelName = array_values(array_filter(
+					[
+						$channel['name'],
```

Comment:
The name is most likely set but isn't it still safer to make it like that as well?
```suggestion
						$channel['name'] ?? null,
```

Reply:
ah I thought it would return null without any error